cmd/cloud-controller-manager/main.go

```diff
@@ -20,27 +20,70 @@ limitations under the License.
 package main
 
 import (
-	"k8s.io/cloud-provider-gcp/cmd/cloud-controller-manager/app"
 	"math/rand"
 	"os"
 	"time"
 
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cloud-provider"
+	"k8s.io/cloud-provider/app"
+	"k8s.io/cloud-provider/app/config"
+	"k8s.io/cloud-provider/options"
+	cliflag "k8s.io/component-base/cli/flag"
 	"k8s.io/component-base/logs"
-	_ "k8s.io/component-base/metrics/prometheus/version" // for version metric registration
-	// NOTE: Importing all in-tree cloud-providers is not required when
-	// implementing an out-of-tree cloud-provider.
-	_ "k8s.io/cloud-provider-gcp/providers/gce"
 	_ "k8s.io/component-base/metrics/prometheus/clientgo" // load all the prometheus client-go plugins
+	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
+	"k8s.io/klog/v2"
+
+	_ "k8s.io/cloud-provider-gcp/providers/gce"
+	// For existing cloud providers, the option to import legacy providers is still available.
+	// e.g. _"k8s.io/legacy-cloud-providers/<provider>"
 )
 
 func main() {
 	rand.Seed(time.Now().UnixNano())
+
+	ccmOptions, err := options.NewCloudControllerManagerOptions()
+	if err != nil {
+		klog.Fatalf("unable to initialize command options: %v", err)
+	}
+
+	fss := cliflag.NamedFlagSets{}
+	command := app.NewCloudControllerManagerCommand(ccmOptions, cloudInitializer, app.DefaultInitFuncConstructors, fss, wait.NeverStop)
+
+	// TODO: once we switch everything over to Cobra commands, we can go back to calling
```

Comment:
Can we have a bug referenced for this TODO?

Reply:
It was been added by https://github.com/kubernetes/kubernetes/pull/58407. And together within some other files(like `https://github.com/kubernetes/kubernetes/blob/271476446b32bfc69a95ea20d8b02d5303128f5e/cmd/kube-proxy/proxy.go#L39`). I will create an issue for tracking.

Reply:
Can we have a separate issue for this repo to fix it here. Then reference that issue in this TODO?

Reply:
Updated. Ref: https://github.com/kubernetes/cloud-provider-gcp/issues/215